lib/internal/Magento/Framework/Code/Reader/NamespaceResolver.php

```diff
@@ -104,8 +105,14 @@ public function getImportedNamespaces(array $fileContent)
                     $import = array_filter(
                         $import,
                         function ($token) {
-                            $whitelist = [T_NS_SEPARATOR, T_STRING, T_AS];
-                            if (isset($token[0]) && in_array($token[0], $whitelist)) {
+                            $whitelist = [
+                                T_NS_SEPARATOR => T_NS_SEPARATOR,
+                                T_STRING => T_STRING,
+                                T_AS => T_AS,
+                                T_NAME_QUALIFIED => T_NAME_QUALIFIED,
+                                T_NAME_FULLY_QUALIFIED => T_NAME_FULLY_QUALIFIED
+                            ];
+                            if (isset($token[0]) && \array_key_exists($token[0], $whitelist)) {
```

Comment:
Any reasons of not using `isset` instead of `array_key_exists`?
Something like that.
```suggestion
                            if (isset($token[0]) && isset($whitelist[$token[0]])) {
```

Reply:
it doesn't make any sense since php 7.4 
https://github.com/magento/magento2/pull/34062#discussion_r708360426

Reply:
https://github.com/magento/magento2/pull/34062#discussion_r708368985